drivers/serial/Kconfig.pl011

```diff
@@ -4,7 +4,7 @@
 menuconfig UART_PL011
 	bool "ARM PL011 UART Driver"
 	select SERIAL_HAS_DRIVER
-	select SERIAL_SUPPORT_INTERRUPT
+	select SERIAL_SUPPORT_INTERRUPT if $(dt_compat_enabled,$(DT_PL011_PORT0_IRQ_TX))
```

Comment:
Regardless of whether a platform uses UART interrupts or not, it is true that PL011 UART hardware supports interrupts and therefore `SERIAL_SUPPORT_INTERRUPT` should be selected.

If you are not using interrupts, you should make sure that `UART_INTERRUPT_DRIVEN` is not selected by the SoC.

Reply:
> useful for example when QEMU where 4 different non-shared IRQs are not supported

As for this, you should instantiate a shared IRQ "interrupt controller" in the dts and configure the interrupt parent of UART as that shared IRQ interrupt controller (if you are ever going to use UART IRQs, that is).

Reply:
> If you are not using interrupts, you should make sure that `UART_INTERRUPT_DRIVEN` is not selected by the SoC.

Sometimes this is not enough. See for example `samples/subsys/console/echo/sample.yaml`, the filter is only on `CONFIG_UART_CONSOLE and CONFIG_SERIAL_SUPPORT_INTERRUPT` so even if your platform has `UART_INTERRUPT_DRIVEN=n` the sample is compiled and it fails with `error: ‘DT_PL011_PORT0_IRQ_TX’ undeclared`

If we do not want to fix the way I did here we have to fix all the tests filtering also on `UART_INTERRUPT_DRIVEN`.

Reply:
The `filter` in that `sample.yaml` should really be `CONFIG_UART_CONSOLE and CONFIG_UART_INTERRUPT_DRIVEN`, as `CONFIG_UART_INTERRUPT_DRIVEN` is actually what dictates whether UART interrupts are used or not (and console subsys tty depends on it).

Also did you do `UART_INTERRUPT_DRIVEN=n` or `CONFIG_UART_INTERRUPT_DRIVEN=n`? If you did the latter, `error: ‘DT_PL011_PORT0_IRQ_TX’ undeclared` should not have occurred.

Reply:
> Also did you do `UART_INTERRUPT_DRIVEN=n` or `CONFIG_UART_INTERRUPT_DRIVEN=n`? If you did the latter, `error: ‘DT_PL011_PORT0_IRQ_TX’ undeclared` should not have occurred.

I think that it is being forced by: `prj.conf` -> `CONSOLE_GETCHAR` -> `CONSOLE_HANDLER`-> `UART_INTERRUPT_DRIVEN`

Reply:
Given that `UART_INTERRUPT_DRIVEN` is being unconditionally set by the `select` chain we cannot solve this filtering on `UART_INTERRUPT_DRIVEN` in `sample.yaml` (the symbol is always set to `=y`).

I'm going to exclude this test for my platform (or enable the shared IRQs). Going to resubmit this patches without this commit.